sign/rpmgensig.c

```diff
@@ -317,11 +347,13 @@ static int runGPG(sigTarget sigt, const char *sigfile)
 
 exit:
 
+    sigaction(SIGCHLD, &oldact, NULL);
+
     if (fnamedPipe)
 	Fclose(fnamedPipe);
 
-    if (pid)
-	waitpid(pid, &status, 0);
+    if (gpgPid)
+	waitpid(gpgPid, &status, 0);
 
```

Comment:
Not a problem introduced in this patch so not something you need to address, just noting: the exit section duplicates much of what is being done just above it. The whole point of having exit/error labels to jump to is to avoid just that...

Reply:
Hopefully resolved in the second commit :)